jetty-start/src/main/java/org/eclipse/jetty/start/CommandLineBuilder.java

```diff
@@ -180,7 +158,32 @@ public String toString(String delim)
             {
                 buf.append(delim);
             }
-            buf.append(quote(arg));
+            buf.append(arg); // we assume escaping has occurred during addArg
+        }
+
+        return buf.toString();
+    }
+
+    /**
+     * A version of {@link #toString()} where every arg is evaluated for potential {@code '} (single-quote tick) wrapping.
+     *
+     * @param delim the delimiter between args, use {@code ' '} (space) for shell executable command line.
+     * @return the toString but each arg that has spaces is surrounded by {@code '} (single-quote tick)
+     */
+    public String toQuotedString(String delim)
+    {
+        StringBuilder buf = new StringBuilder();
+
+        for (String arg : args)
+        {
+            if (buf.length() > 0)
+                buf.append(delim);
+            boolean needsQuotes = (arg.contains(" "));
```

Comment:
should this not be:
```suggestion
            boolean needsQuotes = (arg.contains(delim));
```

Reply:
Not really, as this is testing for a space present in a single arg.
`delim`, as used above, is the delim between args. 

See the Main.java, the `delim` you see as a method parameter is different depending on if debug enabled or not (with debug it's `\n` (LF), otherwise its ` ` (space))

Reply:
But isn't it true that you need to quote an arg only if it contains a delimiter?   If space is not the delimiter, then you don't need to quote it.